### ISSUE TITLE
model type does not use component name

### ISSUE THREAD

**Issue by rattkin** [ferdikoomen/openapi-typescript-codegen#408]
I have this openapi:
```
openapi: 3.0.0
info:
    title: testString
    version: 1.0.0
    description: ""
components:
    schemas:
        Date:
            type: string
            example: 2017-07-21
            description: full-date notation as defined by RFC 3339, section 5.6
        
```

And this is the generated model string.ts:
```
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

/**
 * full-date notation as defined by RFC 3339, section 5.6
 */
export type string = string;
```

Besides being nonsensical, it also can not by compiled by typescript with error message:
`error TS2457: Type alias name cannot be 'string'.`

As a side note, ng-openapi-gen package does it IMHO correctly:
```
export type Date = string;
```

**Comment by rattkin** [comment]
the problem disappears when i rename the schema name Date to something else. (lowercase date also has this problem)

**Comment by rattkin** [comment]
I realize having schema name same as a type name is a bad idea and I changed our api spec. However naming it 'number' generates correctly.

**Comment by ferdikoomen** [comment]
@rattkin I'll fix the issue, but keep in mind that a model called Date will conflict with the default Date class in Javascript.. So it might not be very developer friendly. And if Date is a string, why does the REST API not simply define date fields as string, rather then exposing a type-only model called Date? Anyway, that is a different discussion. I'll push a fix soon and upload a new version: 0.6.0 after that.

**Comment by rattkin** [comment]
Thank you!
